packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -545,27 +541,6 @@ async function createComponentTreeInternal({
   // We avoid cloning this object because it gets consumed here exclusively.
   const props: { [prop: string]: any } = parallelRouteProps
 
-  // If it's a not found route, and we don't have any matched parallel
-  // routes, we try to render the not found component if it exists.
-  if (
-    NotFound &&
-    asNotFound &&
-    // In development, it could hit the parallel-route-default not found, so we only need to check the segment.
-    // Or if there's no parallel routes means it reaches the end.
-    !parallelRouteMap.length
-  ) {
-    props.children = (
-      <>
-        <meta name="robots" content="noindex" />
```

Comment:
we're removing this `<meta name="robots" content="noindex" />` here, is it being added anywhere else? seems important for a 404 page

Reply:
it's managing in the app render with `NonIndex` metadata, if status is 404 it will render the metadata for you there. Tests are covered in `test/e2e/app-dir/not-found` and `test/e2e/app-dir/navigation`